sky/adaptors/aws.py

```diff
@@ -56,7 +57,13 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def _assert_kwargs_builtin_type(kwargs):
+    assert all(isinstance(v, (int, float, str)) for v in kwargs.values()), (
+        f'kwargs should not contain none built-in types: {kwargs}')
+
+
 @import_package
+@functools.lru_cache()
```

Comment:
For discussion: I feel lru_cache() is harder to reason about than using thread local. For example, it raises questions about things like L87 that prompted us to add thread local in the first place.

How about reverting to the thread local solution?

Reply:
An LRU cache is needed with `maxsize` involved, as our original thread local is prone to the error for unbounded memory consumption, and that is why we got into this serious issue of OOM.

I found it much more readable and cleaner with the `lru_cache` instead of the self-implemented LRUCache.

I am not sure why the thread local implementation is better for raising the error L87. Could you say more? It feels exactly the same for reasoning the multi-thread issue to me than implementing a customized LRU, but cleaner in code.